x/mongo/driver/topology/connection_test.go

```diff
@@ -485,6 +485,41 @@ func TestConnection(t *testing.T) {
 					}
 					listener.assertMethodsCalled(t, 1, 1)
 				})
+				t.Run("message too large errors", func(t *testing.T) {
+					testCases := []struct {
+						name   string
+						buffer []byte
+						desc   description.Server
+					}{
+						{
+							"message too large errors with small max message size",
+							[]byte{0x0A, 0x00, 0x00, 0x00, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A},
```

Comment:
Can this be just the first four bytes for the length? With this change, we should be ensuring we do not attempt to read beyond the first four bytes of the length if it exceeds the max message size.

Reply:
Sure!